File: Web2Proj/src/app/shared/flight-service-and-model/flight.ts
import { AirCompanyProfileComponent } from 'src/app/components/air-company-profile/air-company-profile.component'
import { AirCompany } from 'src/app/entities/airCompany'
import { Seat } from 'src/app/entities/seat'
import { Destination } from 'src/app/entities/destination'
import { ServiceGrade } from 'src/app/entities/service-grade'
import { HashLocationStrategy } from '@angular/common'

export class Flight {

    FlightID: number
    DepartureTime: Date
    LandingTime: Date
    TravelTime: string
    TravelLength: number
    TicketPrice: number
    NumberOfChangeovers: number
    TicketDisctount: boolean
    NewTicketPrice: number
    AirCompany: AirCompany
    ServiceGrades: ServiceGrade
   
}

File: Web2Backend/Web2Backend/Model/Flight.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Threading.Tasks;
using Web2Backend.JoiningTableModels;

namespace Web2Backend.Model
{
    public class Flight
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int FlightID { get; set; }
        public DateTime DepartureTime { get; set; }
        public DateTime LandingTime { get; set; }
        public string TravelTime { get; set; }
        public int TravelLength { get; set; }
        public int TicketPrice { get; set; }
        public int NumberOfChangeovers { get; set; }

        public bool TicketDisctount { get; set; }
        [AllowNull]
        public int NewTicketPrice { get; set; }

        public AirCompany AirCompany { get; set; }
        public virtual ICollection<Seat> Seats { get; set; }
        public virtual ICollection<FlightDestinations> FlightDestinations { get; set; }//Presedanja
        public virtual ICollection<ServiceGrade> ServiceGrades { get; set; }
        public Flight()
        {
            this.Seats = new HashSet<Seat>();
            this.FlightDestinations = new HashSet<FlightDestinations>();
            this.ServiceGrades = new HashSet<ServiceGrade>();
        }
    }
}
